dot/import_integration_test.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+//go:build integration
+// +build integration
+
+package dot
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"math/big"
+	"os"
+	"testing"
+
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/utils"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+
+	"github.com/stretchr/testify/require"
+)
+
+func setupStateFile(t *testing.T) string {
+	const filename = "../lib/runtime/test_data/kusama/block1482002.out"
+
+	data, err := ioutil.ReadFile(filename)
+	require.NoError(t, err)
+
+	rpcPairs := make(map[string]interface{})
+	err = json.Unmarshal(data, &rpcPairs)
+	require.NoError(t, err)
+	pairs := rpcPairs["result"].([]interface{})
+
+	bz, err := json.Marshal(pairs)
+	require.NoError(t, err)
+
+	fp := "./test_data/state.json"
+	err = ioutil.WriteFile(fp, bz, 0777)
+	require.NoError(t, err)
+
+	return fp
+}
+
+func setupHeaderFile(t *testing.T) string {
+	t.Helper()
+
+	const headerStr = "{\"digest\":{\"logs" +
+		"\":[\"0x0642414245b501013c0000009659bd0f0000000070edad1c9064fff78cb18435223d8ada" +
+		"f5ea04c24b1a8766e3dc01eb03cc6a0c11b79793d4e31cc0990838229c44fed1669a7c7c79e1e6d0a96374d6496728069d1ef739e290" +
+		"497a0e3b728fa88fcbdd3a5504e0efde0242e7a806dd4fa9260c\"," +
+		"\"0x054241424501019e7f28dddcf27c1e6b328d5694c368d5b2ec5dbe0e412ae1c98f88d53be4d8502fac571f3f19c9caaf281a6733" +
+		"19241e0c5095a683ad34316204088a36a4bd86\"]}," +
+		"\"extrinsicsRoot\":\"0xda26dc8c1455f8f81cae12e4fc59e23ce961b2c837f6d3f664283af906d344e0\"," +
+		"\"number\":\"0x169d12\",\"parentHash\":\"0x3b45c9c22dcece75a30acc9c2968cb311e6b0557350f83b430f47559db786975" +
+		"\", \"stateRoot\":\"0x09f9ca28df0560c2291aa16b56e15e07d1e1927088f51356d522722aa90ca7cb\"}"
```

Comment:
next time you can use the backtick ` to surround the string in order to not have to escape double quotes "